packages/domain-picker/src/hooks/use-domain-suggestions.ts

```diff
@@ -1,35 +1,64 @@
 /**
  * External dependencies
  */
-import { useSelect } from '@wordpress/data';
+import { useSelect, useDispatch } from '@wordpress/data';
+import type { DataState } from '@automattic/data-stores/src/domain-suggestions/constants';
+import type { DomainSuggestion } from '@automattic/data-stores/src/domain-suggestions/types';
 import { useDebounce } from 'use-debounce';
 
 /**
  * Internal dependencies
  */
 import { DOMAIN_SUGGESTIONS_STORE, DOMAIN_SEARCH_DEBOUNCE_INTERVAL } from '../constants';
 
+type ReturnType = {
```

Comment:
I would give this a more specific name.

```suggestion
type DomainSuggestionsResult = {
```

Reply:
also this is a mapped type provided by TypeScript so it's bound to confuse people.

```ts
type MyBool = boolean;
type BoolProducer = () => boolean;

// MyBool === ReturnType<BoolProducer>
```

that code works up to the point that `ReturnType` becomes its own concrete type, then who knows - a bunch of errors